src/Microsoft.AspNetCore.Routing/Matching/VectorizedRefEmitTrieJumpTable.cs

```diff
@@ -0,0 +1,548 @@
+
+        private readonly int _defaultDestination;
+        private readonly int _exitDestination;
+        private readonly (string text, int destination)[] _entries;
+
+        private readonly Func<string, int, int, int> _delegate;
+
+        public VectorizedRefEmitTrieJumpTable(
+            int defaultDestination,
+            int exitDestination,
+            (string text, int destination)[] entries)
+        {
+            _defaultDestination = defaultDestination;
+            _exitDestination = exitDestination;
+            _entries = entries;
+
+            _delegate = GenerateDelegate();
+        }
+
+        public override int GetDestination(string path, PathSegment segment)
+        {
+            if (path.Length == 0)
+            {
+                return _exitDestination;
+            }
+
+            return _delegate(path, segment.Start, segment.Length);
+        }
+
+        private Func<string, int, int, int> GenerateDelegate()
+        {
+#if NET462
+            var assemblyName = new AssemblyName("DynamicRoutingJumpTable");
+            var assembly = AppDomain.CurrentDomain.DefineDynamicAssembly(assemblyName, AssemblyBuilderAccess.RunAndSave);
+            var module = assembly.DefineDynamicModule(assemblyName.Name, assemblyName.Name + ".dll");
+            var type = module.DefineType("Test", TypeAttributes.Public);
+            var method = type.DefineMethod(
+                "GeneratedGetDestination",
+                MethodAttributes.Public | MethodAttributes.Static,
+                typeof(int),
+                new[] { typeof(string), typeof(int), typeof(int), });
+#else
+            var method = new DynamicMethod(
+                "GeneratedGetDestination",
+                typeof(int),
+                new [] { typeof(string), typeof(int), typeof(int), });
+
+#endif
+            var il = method.GetILGenerator();
+            
+            // See comments on Locals for details
+            var locals = new Locals()
+            {
+                P = il.DeclareLocal(typeof(byte).MakeByRefType()),
+
+                UInt16Value = il.DeclareLocal(typeof(ushort)),
+
+                UInt64Value = il.DeclareLocal(typeof(ulong)),
+                UInt64LowerIndicator = il.DeclareLocal(typeof(ulong)),
+                UInt64UpperIndicator = il.DeclareLocal(typeof(ulong)),
+
+                Span = il.DeclareLocal(typeof(ReadOnlySpan<char>)),
+            };
+
+            // See comments on Labels for details
+            var labels = new Labels()
+            {
+                ReturnDefault = il.DefineLabel(),
+                ReturnNotAscii = il.DefineLabel(),
+            };
+
+            // See comments on Methods for details
+            var methods = new Methods();
+
+            // Initializing top-level locals - this is similar to...
+            // ReadOnlySpan<char> span = arg0.AsSpan(arg1, arg2);
+            // ref byte p = ref Unsafe.As<char, byte>(MemoryMarshal.GetReference<char>(span))
```

Comment:
I imagine it's the same code internally, but imho ref var b = ref MemoryMarshal.GetReference(MemoryMarshal.AsBytes(name)); is more readable.

Reply:
```csharp
ref MemoryMarshal.GetReference(MemoryMarshal.AsBytes(name));
```
Would first get the reference to the first element of span, `Unsafe.As` that then [create a new Span](https://github.com/dotnet/coreclr/blob/b2031a14a3b110f8e712abdb0cd392c54cd2e839/src/System.Private.CoreLib/shared/System/Runtime/InteropServices/MemoryMarshal.Fast.cs#L28) and then get the reference of the first element of that:
```csharp
public static Span<byte> AsBytes<T>(Span<T> span)
    where T : struct
{
    if (RuntimeHelpers.IsReferenceOrContainsReferences<T>())
        ThrowHelper.ThrowInvalidTypeWithPointersNotSupported(typeof(T));

    return new Span<byte>(
        ref Unsafe.As<T, byte>(ref GetReference(span)),
        checked(span.Length * Unsafe.SizeOf<T>()));
}
```
So would be 2x `GetReference`; 1x `Unsafe.As`; 1x `new Span<byte>`

Current is 1x `GetReference`; 1x `Unsafe.As`?

Reply:
Urgh, ok, forget that then, bad idea.